bundles/core/src/test/java/com/adobe/cq/commerce/core/components/internal/models/v1/teaser/CommerceTeaserImplTest.java

```diff
@@ -0,0 +1,122 @@
+
+    @Before
+    public void setup() {
+        CommerceTeaserImpl cifTeaser = new CommerceTeaserImpl();
+
+        Page productPage = mock(Page.class);
+        Page categoryPage = mock(Page.class);
+
+        when(productPage.getPath()).thenReturn(productPath);
+        when(categoryPage.getPath()).thenReturn(categoryPath);
+
+        List<Resource> actionResources = new ArrayList<>();
+
+        // Action node 1 - productSKU configured, category left blank, text set to some value
+        actionResources.add(getActionNodeResource("278", null, "My Product"));
+
+        // Action node 2 - productSKU left blank, category configured, text set to some value
+        actionResources.add(getActionNodeResource(null, "30", "My Category"));
+
+        // Action node 3 - productSKU configured, category also configured, text set to some value
+        actionResources.add(getActionNodeResource("278", "30", "My Category"));
+
+        // Action node 4 - productSKU left blank, category also left blank, text set to some value
+        actionResources.add(getActionNodeResource(null, null, "This Page"));
+
+        Resource mockedResource = mock(Resource.class);
+        Map<String, Object> actionProperties = new HashMap<>();
+
+        actionProperties.put(CommerceTeaser.PN_ACTIONS_ENABLED, true);// Setting action enabled TRUE
+
+        when(mockedResource.getValueMap()).thenReturn(new ValueMapDecorator(actionProperties));
+
+        Resource mockedChildResource = mock(Resource.class);
+
+        when(mockedChildResource.getChildren()).thenReturn(actionResources);
+        when(mockedResource.getChild(CommerceTeaser.NN_ACTIONS)).thenReturn(mockedChildResource);
+
+        Whitebox.setInternalState(cifTeaser, "resource", mockedResource);
+        Whitebox.setInternalState(cifTeaser, "categoryPage", categoryPage);
+        Whitebox.setInternalState(cifTeaser, "productPage", productPage);
+        // Whitebox.setInternalState(cifTeaser, "actionsEnabled", true);
+
+        cifTeaser.setActionsEnabled();
+        cifTeaser.populateActions();
+        this.slingModel = cifTeaser;
+
+    }
+
+    Resource getActionNodeResource(String sku, String categoryId, String text) {
+        Resource actionResource = mock(Resource.class);
+
+        Map<String, Object> actionProperties = new HashMap<>();
+
+        actionProperties.put(CommerceTeaser.PN_ACTION_PRODUCT_SKU, sku);
+        actionProperties.put(CommerceTeaser.PN_ACTION_CATEGORY_ID, categoryId);
+        actionProperties.put(CommerceTeaser.PN_ACTION_TEXT, text);
+
+        ValueMapDecorator vMD = new ValueMapDecorator(actionProperties);
+
+        when(actionResource.getValueMap()).thenReturn(vMD);
+
+        return actionResource;
+    }
+
+    @Test
+    public void verifyActions() {
+
+        List<ListItem> actionItems = this.slingModel.getActions();
+
+        Assert.assertTrue(this.slingModel.isActionsEnabled());
+        Assert.assertTrue(actionItems.size() == 4);
```

Comment:
You add 4 actions in the test, but you use asserts only for 3 of them. What about the 4th case?

Reply:
here, the behavior is that if niether productSKU nor CategoryID is defined, then Link will point to `currentPage` , skipped this in testing as we are not setting `currentPage` in slingModel mocks in the test class - do you think it should also be asserted?

Reply:
If you don't assert it, why do you add it here https://github.com/adobe/aem-core-cif-components/pull/74/files/b22fa76672e0b9773b0a52a966bfa832f65a5831#diff-cb87fd5300d137fc2056c8ad9a1a7224R63 then?

Reply:
Did so to verify that even null ProductSKU and null CategoryId with a title is also getting added in action arraylist verified through `Assert.assertTrue(actionItems.size() == 4)` , I will add a mocked currentPage as well, just to cover 4rth assertion too. Thanks.

Reply:
well what do you know, the added test case to realize its failing - fixed it in actionItem class.
this is done.

Reply:
Good job 👍